src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -397,5 +399,37 @@ public bool DetermineExtensionDataProperty(Dictionary<string, JsonPropertyInfo>
 
             return converter;
         }
+
+        public static void ValidateType(Type type, Type? parentClassType, PropertyInfo? propertyInfo, JsonSerializerOptions options)
+        {
+            if (!options.TypeIsCached(type) && IsInvalidForSerialization(type))
+            {
+                ThrowHelper.ThrowInvalidOperationException_CannotSerializeInvalidType(type, parentClassType, propertyInfo);
+            }
+        }
+
+        public static bool IsInvalidForSerialization(Type type)
```

Comment:
I don't think any of these helpers need to be public. Update access modifiers across the board to be private/internal where it makes sense and only keep thing public where they are necessary. I understand `JsonClassInfo` is internal, but we should still try to have a clean separation of concerns even within the intenal impls.

Reply:
Yes. This should have been cleaned up following iteration on this PR.